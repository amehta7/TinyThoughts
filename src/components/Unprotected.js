import React, { memo } from "react";

const Unprotected = memo(() => {
  return <></>;
});

export default Unprotected;
