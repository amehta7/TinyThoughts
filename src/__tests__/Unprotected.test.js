import React from "react";
import { mount } from "enzyme";
import { MemoryRouter, Redirect as MockRedirect } from "react-router-dom";
import { Provider } from "react-redux";
import configureStore from "redux-mock-store";
import thunk from "redux-thunk";
import Unprotected from "../components/Unprotected";
const mockStore = configureStore([thunk]);

jest.mock("react-router-dom", () => {
  const RouterMocks = jest.requireActual("react-router-dom");
  return {
    ...RouterMocks,
    Redirect: jest.fn().mockImplementation(() => <span>REDIRECTED</span>)
  };
});

afterEach(() => {
  MockRedirect.mockClear();
});

describe("Unprotected Component", () => {
  it("Renders components when signed out", () => {
    const store = mockStore({
      users: {}
    });

    const wrapper = mount(
      <Provider store={store}>
        <MemoryRouter keyLength={0}>
          <Unprotected>
            <h1>DEMO CONTENT</h1>
          </Unprotected>
        </MemoryRouter>
      </Provider>
    );

    return expect(wrapper.find(Unprotected)).toMatchSnapshot();
  });

  it("Redirects to root route when signed in", () => {
    const store = mockStore({
      users: { user: { id: 1, name: "Joe Mockery" } }
    });

    const wrapper = mount(
      <Provider store={store}>
        <MemoryRouter>
          <Unprotected>
            <h1>DEMO CONTENT</h1>
          </Unprotected>
        </MemoryRouter>
      </Provider>
    );

    expect(wrapper.find(Unprotected).text()).toBe("REDIRECTED");
  });
});
