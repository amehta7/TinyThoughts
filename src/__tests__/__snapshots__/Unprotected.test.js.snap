// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Unprotected Component Renders components when signed out 1`] = `
<Memo()>
  <Route
    render={[Function]}
  >
    <h1>
      DEMO CONTENT
    </h1>
  </Route>
</Memo()>
`;
